tests/all.rs

```diff
@@ -1183,6 +1183,48 @@ fn sparse_with_trailing() {
     assert_eq!(&s[0x100_000..], "1MB through\n");
 }
 
+#[test]
+#[cfg(any(
+    target_os = "dragonfly",
+    target_os = "freebsd",
+    target_os = "illumos",
+    target_os = "linux",
+    target_os = "solaris"
+))]
+fn writing_sparse() {
+    use std::os::unix::prelude::*;
+
+    let mut ar = Builder::new(Vec::new());
+    let td = t!(TempBuilder::new().prefix("tar-rs").tempdir());
+
+    let path = td.path().join("test");
+    let mut file = t!(File::create(&path));
+    t!(file.seek(io::SeekFrom::Start(0x80_000)));
+    t!(file.write_all(b"hello\n"));
+    t!(file.seek(io::SeekFrom::Start(0x100_000)));
+    t!(file.write_all(b"world\n"));
+    let meta = t!(file.metadata());
+    assert!(meta.blocks() <= 32, "can't create sparse file for testing");
+    drop(file);
+
+    t!(ar.append_file("test", &mut t!(File::open(&path))));
+    t!(ar.finish());
+
+    let data = t!(ar.into_inner());
+    assert!(data.len() <= 0x4_000); // Typically 0x1800 but potentially might
+                                    // vary based on a filesystem.
```

Comment:
Could this entire test run unconditionally on all systems, and this assert be limited to only the known OSes that implement sparse file support?